virttest/utils_v2v.py

```diff
@@ -976,12 +1030,71 @@ def get_vddk_thumbprint(host, password, uri_type=None, prompt=r"[\#\$\[\]]"):
     else:
         cmd = 'openssl x509 -in /etc/vmware-vpx/ssl/rui.crt -fingerprint -sha1 -noout'
 
-    r_runner = remote.RemoteRunner(host=host,
-                                   password=password,
-                                   prompt=prompt,
-                                   preferred_authenticaton='password,keyboard-interactive')
+    r_runner = remote.RemoteRunner(
+        host=host,
+        password=password,
+        prompt=prompt,
+        preferred_authenticaton='password,keyboard-interactive')
     cmdresult = r_runner.run(cmd)
     logging.debug("vddk thumbprint:\n%s", cmdresult.stdout)
     vddk_thumbprint = cmdresult.stdout.strip().split('=')[1]
 
     return vddk_thumbprint
+
+
+def setup_esx_ssh_key(hostname, username, password, port=22):
+    """
+    Setup up remote login in esx server by using public key
+
+    :param host: hostname or IP address
+    :param user: username
+    :param password: Password
+    :param port: ssh port number
+    """
+    session = None
+    logging.debug('Performing SSH key setup on %s:%d as %s.' %
+                  (hostname, port, username))
+    try:
+        session = remote.remote_login(
+            client='ssh',
+            host=hostname,
+            username=username,
+            port=port,
+            password=password,
+            prompt=r"[\#\$\[\]]",
+            preferred_authenticaton='password,keyboard-interactive')
+        public_key = ssh_key.get_public_key()
+        session.cmd("echo '%s' >> /etc/ssh/keys-root/authorized_keys; " %
+                    public_key)
+        logging.debug('SSH key setup complete.')
+
+        return public_key
+    except Exception as err:
+        logging.debug('SSH key setup has failed: %s', err)
```

Comment:
It looks like nothing will happen if ssh key setup failed. The var 'pub_key' will get 'None' as value in line:222 and then the 'split' function will get error in line:224.

Reply:
Sorry, not the 'split' but the index '[1]'

Reply:
That may happen when remote-login has problem, and pub_key will be 'None', and then will get an exception and break the scripts. I think this is fine because it breaks at where we hope it stops running.
We can keep it and see if has other problems.
Thanks.

Reply:
I see. But the error message will be 'index error', it's quite not clear for this error and will be easily marked as 'similar reason' with other failed cases.
Giving a clear error message is the least we can do, especially when we already realize the risk.

Reply:
Thanks @chloerh , that makes sense.  I raise a TestFail exception to exit the script for that.